File: backend/aspnet/Models/Order.cs
namespace aspnet.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

public class Order
{
  public int Id { get; set; }

  public int? ViewerId { get; set; }
  [ForeignKey(nameof(ViewerId))]
  public Viewer Viewer { get; set; }

  public int? TicketId { get; set; }
  [ForeignKey(nameof(TicketId))]
  public Ticket Ticket { get; set; }

  [Required]
  public DateTime OrderDate { get; set; }
}

File: backend/express/models/order.js
const mongoose = require('mongoose');

const orderSchema = new mongoose.Schema({
  viewerId: { type: mongoose.Schema.Types.ObjectId, ref: 'Viewer', required: true },
  ticketId: { type: mongoose.Schema.Types.ObjectId, ref: 'Ticket', required: true },
  sellerId: { type: mongoose.Schema.Types.ObjectId, ref: 'Seller', required: true },
  orderDate: { type: Date, required: true },
});

module.exports = mongoose.model('Order', orderSchema);
